### DISCUSSION TITLE
Make theme generator more user friendly

### DISCUSSION THREAD

**Discussion by sovetski** [saadeghi/daisyui#3017] [Ideas / feature requests]
Recently the `optional colors` support added which is very good, but for me the theme generator is not user friendly.

My suggestions are:

- [x] Add copy button to copy the JSON to the clipboard, it is very anoying to select manually each time
- [x] Add import/export feature to be able to save as a json file easily. Export feature can be ignored maybe if we have copy one like above, but the import will be very useful. Because some times we have to test again with some small changes and writing manually is difficult
- [x] Add border size config too like `optional colors`, for example in `business` theme border radius is different, and if we extend from this theme, we can not customize the border radius and see the real changes with theme generator
- [x] Share feature is one of helpful feature for me, if we want to show to our client the colors before applying it to the website, we can just make some config and share the URL with the client like `https://daisyui.com/theme-generator/share/xxxxx` and then the client can make another changes and share a new link etc.
- [x] Ability to select existing themes colors and once we selected them, the default colors are automatically imported. I mean, imagine we don't want to create a theme from scratch and we already love some existing theme colors. If we have a selectbox with all existing default themes, by selecting them we can get all colors prefilled, so then we can adapt it as we want. In my case, I checked the source code and found this one to see what are the default colors: https://github.com/saadeghi/daisyui/blob/master/src/theming/themes.js

**Comment by Billdex** [comment]
That's a useful suggestion. Some browser can't use oklch color and make our website seem terrible on some users's browser client. Try to rebuild a new theme which seem like a existing theme is also inconvenient. Include existing default themes will be helpful !

**Comment by ericridgeway** [comment]
We desperately need the ability to re-Import back to the theme-generator so we can continue to make tweaks

**Reply by saadeghi** [reply]
You can import into the "CSS" modal. If format is valid it will work (if not, look into devtools for validatione errors)

**Comment by ericridgeway** [comment]
Oh, that worked! Thank you! :)

**Comment by sovetski** [comment]
From what I see, all mentioned features are implemented. Thanks, so I am closing this.
